### PR TITLE
WIP Initial checkin of rpc

### PR DESCRIPTION
**Pull request by carnesen**
I messed up the rebase on that other unmerged branch to I'm trying again on a "clean" rpc branch. (Also including "protocol" in the BitcoindRpc constructor options as per Manuel's feedback.)

### CODE REVIEW

**Comment by maraoz** [comment]
This looks good as a rough skeleton of something, but I still don't see what the `BitcoindRpc` class will be used for in the app. Feel free to update this PR with more work and we'll consider merging it :)

btw, great work respecting our code style :+1:

File: lib/Rpc.js [hunk 1]
```diff
@@ -1,119 +1,25 @@
...
-    return cb(null,info);
-  });
+var bitcore = require('bitcore');
+var $ = bitcore.util.preconditions;
+var RpcClient       = require('bitcoind-rpc');
+var util            = require('util');
+
+var BitcoindRpc = function(eventBus, client) {
+  $.checkArgument(eventBus);
+  $.checkArgument(client);
```
Comment (maraoz) [inline lib/Rpc.js]: You probably want to save the variables `eventBus` and `client` in the object

**Comment by carnesen** [comment]
I added a couple methods to BitcoindRpc. I'm not attached to that name by any means, but I was thinking we'd collect the app-specific rpc calls and handling there. The getTx method gets a Transaction by hash. It'll be one way we can get a transaction for the /tx endpoints. The emitMemoryPool method gets the mempool and emits the transactions on the EventBus. The idea there is that we'll call emitMemoryPool on startup to try to capture transactions that were recently relayed, but missed because bitcore-node was down.

**Comment by maraoz** [comment]
I've added "WIP" to the PR title. Please remove it once you think it's ready for review

**Comment by carnesen** [comment]
I see Esteban is integrating RPC calls directly into the "services" abstraction, which seems pretty cool. Retracting this request ...

**Comment by eordano** [comment]
I had my doubts on splitting those services further and having a "DAO" or "RPCProvider" as a thin wrapper for those, but not sure if it's necessary, do you agree?

**Comment by maraoz** [comment]
Yeah, let's not overcomplicate the architecture too early

**Comment by carnesen** [comment]
I think I was aiming for something like what will some day be bitcore-rpc.

**Pull request closed without merging**